bus-mapping/src/evm/opcodes/stackonlyop.rs

```diff
@@ -145,28 +168,237 @@ mod stackonlyop_tests {
+        );
+    }
+    #[test]
+    fn test_sgt_opcode() {
+        sgt_opcode_impl(0x01, 0x02, 0x00);
+        sgt_opcode_impl(0x01, 0x01, 0x00);
+        sgt_opcode_impl(0x02, 0x01, 0x01);
+    }
+
+    fn and_opcode_impl(a: usize, b: usize, result: usize) {
+        stack_only_opcode_impl::<2, 1>(
+            OpcodeId::AND,
+            bytecode! {
+                PUSH1(b)
+                PUSH1(a)
+                AND
+                STOP
+            },
+            vec![
+                StackOp::new(1, StackAddress(1022), Word::from(a)),
+                StackOp::new(1, StackAddress(1023), Word::from(b)),
+            ],
+            vec![StackOp::new(1, StackAddress(1023), Word::from(result))],
+        );
+    }
+    #[test]
+    fn test_and_operate() {
+        and_opcode_impl(0x01, 0x01, 0x01);
+        and_opcode_impl(0x01, 0x00, 0x00);
+        and_opcode_impl(0x00, 0x00, 0x00);
+    }
+
+    fn or_opcode_impl(a: usize, b: usize, result: usize) {
+        stack_only_opcode_impl::<2, 1>(
+            OpcodeId::OR,
+            bytecode! {
+                PUSH1(b)
+                PUSH1(a)
+                OR
+                STOP
+            },
+            vec![
+                StackOp::new(1, StackAddress(1022), Word::from(a)),
+                StackOp::new(1, StackAddress(1023), Word::from(b)),
+            ],
+            vec![StackOp::new(1, StackAddress(1023), Word::from(result))],
+        );
+    }
     #[test]
-    fn addmod_opcode_impl() {
-        stack_only_opcode_impl::<3, 1>(
-            OpcodeId::ADDMOD,
+    fn test_or_operate() {
+        or_opcode_impl(0x01, 0x01, 0x01.bitor(0x01) as usize);
+        or_opcode_impl(0x00, 0x01, 0x00.bitor(0x01) as usize);
+        or_opcode_impl(0x00, 0x00, 0x00.bitor(0x00) as usize);
+    }
+
+    fn xor_opcode_impl(a: usize, b: usize, result: usize) {
+        stack_only_opcode_impl::<2, 1>(
+            OpcodeId::XOR,
             bytecode! {
-                PUSH3(0xbcdef)
-                PUSH3(0x6789a)
-                PUSH3(0x12345)
-                ADDMOD
+                PUSH1(b)
+                PUSH1(a)
+                XOR
                 STOP
             },
             vec![
-                StackOp::new(1, StackAddress(1021), Word::from(0x12345)),
-                StackOp::new(1, StackAddress(1022), Word::from(0x6789a)),
-                StackOp::new(1, StackAddress(1023), Word::from(0xbcdef)),
+                StackOp::new(1, StackAddress(1022), Word::from(a)),
+                StackOp::new(1, StackAddress(1023), Word::from(b)),
             ],
-            vec![StackOp::new(1, StackAddress(1023), Word::from(0x79bdf))],
+            vec![StackOp::new(1, StackAddress(1023), Word::from(result))],
+        );
+    }
+    #[test]
+    fn test_xor_operate() {
+        xor_opcode_impl(0x01, 0x01, 0x01.bitxor(0x01) as usize);
+        xor_opcode_impl(0x01, 0x00, 0x00.bitxor(0x01) as usize);
+        xor_opcode_impl(0x00, 0x00, 0x00.bitxor(0x00) as usize);
```

Comment:
Considering how small the test numbers are, we could simply put the result directly instead of using `bitxor()`. But it's fine if you want to keep it as is.